src/clarity-angular/layout/tabs/tabs.ts

```diff
@@ -3,35 +3,84 @@
+                    <li role="presentation" class="nav-item">
+                        <button class="btn btn-link nav-link dropdown-toggle" [class.active]="activeTabInOverflow">
+                            <clr-icon shape="ellipsis-horizontal" [class.is-info]="ifOpenService.open"></clr-icon>
+                        </button>
+                    </li>
+                    <clr-tab-overflow-content>
+                        <ng-container #overflowTabLinks></ng-container>
+                    </clr-tab-overflow-content>
+                </div>
+            </ng-container>
         </ul>
+        <!--this is the tab content, rendered outside the ul element above-->
+        <!--TODO: find a better way to achieve this-->
         <ng-container *ngFor="let tab of tabsService.children">
             <ng-template *ngIf="tab.tabContent" [ngTemplateOutlet]="tab.tabContent.templateRef"></ng-template>
         </ng-container>
     `,
-    providers: [ IfActiveService, TabsService ]
+    providers: [ IfActiveService, IfOpenService, TabsService ]
 })
 export class Tabs {
-    constructor(public ifActiveService: IfActiveService, public tabsService: TabsService) {
+    private _prevOverflowTabs: Tab[] = [];
+    @ViewChild("inlineTabLinks", {read: ViewContainerRef}) inlineTabLinksContainerRef: ViewContainerRef;
+    @ViewChild("overflowTabLinks", {read: ViewContainerRef}) overflowTabLinksContainerRef: ViewContainerRef;
+    @ContentChildren(TabLinkDirective, {descendants: true}) tabLinkDirectives: QueryList<TabLinkDirective>;
+
+    constructor(public ifActiveService: IfActiveService,
+                public ifOpenService: IfOpenService,
+                public tabsService: TabsService) {
+    }
+
+    get activeTabInOverflow() {
+        return this.tabsService.overflowTabs.indexOf(this.tabsService.activeTab) > -1;
     }
 
-    get activeTab() {
-        let activeTab: Tab;
-        if (this.tabsService.children.length > 0) {
-            activeTab = this.tabsService.children.find((tab: Tab) => {
-                return tab.active;
-            });
+    ngAfterViewInit() {
+        this.renderTabLinks();
+    }
+
+    ngAfterViewChecked() {
+        if (this.overflowTabsChanged()) {
+            this.renderTabLinks();
+            this._prevOverflowTabs = this.tabsService.overflowTabs;
         }
-        return activeTab;
     }
+
+    overflowTabsChanged() {
+        return (this.tabsService.overflowTabs.length !== this._prevOverflowTabs.length);
+    }
+
+    renderTabLinks() {
+        // Set the ContainerRefs and render the view. One of the refs might have been undefined
+        // earlier, so we always set it before calling renderView here rather than
+        // setting them inside ngAfterViewInit() function.
+        this.tabsService.inlineTabLinksContainerRef = this.inlineTabLinksContainerRef;
+        this.tabsService.overflowTabLinksContainerRef = this.overflowTabLinksContainerRef;
+        this.tabsService.renderView();
```

Comment:
So now that I have read everything, I think you can get away with a much, much simpler solution. Just use:
```html
<ng-container *ngFor="let link of tabLinks">
    <ng-container *ngIf="!link.inOverflow"
        [ngTemplateOutlet]="link.templateRefContainer.template"></ng-container>
</ng-container>
```
and same thing for the inOverflow. No more `@ViewChild`, no more manual creation, ...
And you don't need to keep track of two lists, you're using the same in both and just trusting `*ngIf`.